com.unity.render-pipelines.core/Runtime/Utilities/CoreRenderPipelinePreferences.cs

```diff
@@ -10,80 +15,54 @@ namespace UnityEngine.Rendering
-        public static readonly GUIContent userDefaults = EditorGUIUtility.TrTextContent("Use Defaults");
-    }
+
     public static class CoreRenderPipelinePreferences
     {
-        static bool m_Loaded = false;
+        #region Volumes Gizmo Color
 
-        // Added default Colors so that they can be reverted back to these values
         static Color s_VolumeGizmoColorDefault = new Color(0.2f, 0.8f, 0.1f, 0.5f);
-        static Color s_VolumeGizmoColor = s_VolumeGizmoColorDefault;
-        static readonly Color kPreviewCameraBackgroundColorDefault = new Color(82f / 255.0f, 82f / 255.0f, 82.0f / 255.0f, 0.0f);
+        static Func<Color> GetColorPrefVolumeGizmoColor;
 
-        public static Color volumeGizmoColor
-        {
-            get => s_VolumeGizmoColor;
-            set
-            {
-                if (s_VolumeGizmoColor == value) return;
-                s_VolumeGizmoColor = value;
-                EditorPrefs.SetInt(Keys.volumeGizmoColor, (int)ColorUtils.ToHex(value));
-            }
-        }
+        public static Color volumeGizmoColor => GetColorPrefVolumeGizmoColor();
 
-        public static Color previewBackgroundColor => kPreviewCameraBackgroundColorDefault;
+        #endregion
 
-        static class Keys
-        {
-            internal const string volumeGizmoColor = "CoreRP.Volume.GizmoColor";
-        }
+        #region Preview Camera Background Color
 
-        [SettingsProvider]
-        static SettingsProvider PreferenceGUI()
-        {
-            return new SettingsProvider("Preferences/Colors/SRP", SettingsScope.User)
-            {
-                guiHandler = searchContext =>
-                {
-                    if (!m_Loaded)
-                        Load();
-
-                    Rect r = EditorGUILayout.GetControlRect();
-                    r.xMin = 10;
-                    EditorGUIUtility.labelWidth = 251;
-                    volumeGizmoColor = EditorGUI.ColorField(r, "Volume Gizmo Color", volumeGizmoColor);
-
-                    if (GUILayout.Button(Styles.userDefaults, GUILayout.Width(120)))
-                    {
-                        RevertColors();
-                    }
-                }
-            };
-        }
+        static readonly Color kPreviewCameraBackgroundColorDefault = new Color(82f / 255.0f, 82f / 255.0f, 82.0f / 255.0f, 0.0f);
+        public static Color previewBackgroundColor => kPreviewCameraBackgroundColorDefault;
 
-        static void RevertColors()
-        {
-            volumeGizmoColor = s_VolumeGizmoColorDefault;
-        }
+        #endregion
 
         static CoreRenderPipelinePreferences()
         {
-            Load();
+            GetColorPrefVolumeGizmoColor = RegisterSceneColor("Volume Gizmo", s_VolumeGizmoColorDefault);
         }
 
-        static void Load()
+        static readonly ConcurrentStack<object> s_ColorPref = new ConcurrentStack<object>();
+        internal static Func<Color> RegisterSceneColor(string name, Color defaultColor)
```

Comment:
If this code is meant to be used in other SRP, we need a public access on this (example, how to use it for HDRP decals?) Otherelse, we will need to ammend this file each time or grabb this method with reflection in other SRP packages.